packages/template-retail-react-app/app/hooks/use-active-data.js

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2023, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import {ACTIVE_DATA_ENABLE} from '@salesforce/retail-react-app/app/constants'
+
+const useActiveData = () => {
+    return {
+        async sendViewProduct(category, product, type) {
+            if (ACTIVE_DATA_ENABLE == 0) {
+                return
+            }
+            try {
+                if (dw.ac) {
+                    if (category && category.id) {
+                        console.log('Applying category context')
+                        dw.ac.applyContext({category: category.id})
+                    }
+                    if (product && product.id) {
+                        console.log('Capturing product')
+                        dw.ac._capture({id: product.id, type: type})
+                    }
+                }
+                if (typeof dw.ac._scheduleDataSubmission === 'function') {
+                    dw.ac._scheduleDataSubmission()
+                }
+            } catch (err) {
+                console.error(err)
+            }
+        },
+        async sendViewSearch(searchParams, productSearchResult) {
+            if (ACTIVE_DATA_ENABLE == 0) {
```

Comment:
After converting to `ACTIVE_DATA_ENABLE = true` we can avoid the suboptimal usage of `==` which is not a good comparator in javascript
```suggestion
            if (!ACTIVE_DATA_ENABLE) {
```